### PR TITLE
Fixes to allow scopeTest to build on OS X.

### PR DESCRIPTION
**Pull request by ddunbar**

### CODE REVIEW

File: csbuild/toolchain_gcc.py [hunk 1]
```diff
@@ -39,8 +39,14 @@ def __init__( self ):
 		self._objcAbiVersion = None
 
 		if platform.system() == "Darwin":
-			#TODO: Find the best SDK available instead of hardcoding one.
-			self._sysroot = "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk"
+                        # Get the OS X SDK path via 'xcrun'.
```
Comment (ShadauxCat) [inline csbuild/toolchain_gcc.py]: This looks fine; style note, however: it looks like you have indentation with spaces here. CSBuild's style is to use tabs for indentation. This goes against PEP 8, but internally we prefer tabs over spaces.

If you're interested, you can find CSBuild's style guidelines here: https://docs.google.com/document/d/1uWNFW3YzUTumElMOAOCQFs13r3L3b6pB3GzN2-ElWQs/view# . Not all of CSBuild's code properly follows the guidelines right now, but we're making an effort to ensure all new code contributed to the project does.

**Comment by ShadauxCat** [comment]
As a general note, the current github repository is quite a bit out of date. Since we haven't been getting contributions from outside the core team up till now, we've been a bit lax on pushing things to github and have been working primarily on the perforce workshop. I'll try to get updated code pushed in this weekend; in the meantime, if you'd like to look at the very latest code, you can find it at https://swarm.workshop.perforce.com/projects/shadauxcat-csbuild/ . There's a considerable amount of code targeting MacOS and iOS in that repository that hasn't been properly replicated here yet.

**Comment by ShadauxCat** [comment]
(That said, the contributions you made here are very much needed so I will merge that in with what we have. Thank you!)

**Comment by ddunbar** [comment]
Ah, ok that makes sense, thanks for the link!

 - Daniel

**Comment by ShadauxCat** [comment]
Finally got around to pulling in the p4 workshop changes, your changes should appear in there. They've been merged into the "unsable" branch and once we're ready to release the numerous other changes we've made in the past several weeks, we'll bring it with them into the "master" branch.

**Pull request closed without merging**